custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -252,9 +257,12 @@ def updates(self):
         """.format(
             tablename=self.temporary_tablename,
             ucr_tablename=get_table_name(self.domain, 'static-person_cases_v3'),
+            migration_table=AGG_MIGRATION_TABLE,
             seeking_services=(
                 "CASE WHEN "
-                "registered_status IS DISTINCT FROM 0 AND migration_status IS DISTINCT FROM 1 "
+                "registered_status IS DISTINCT FROM 0 AND "
+                "(agg_migration.is_migrated=1 AND "
```

Comment:
this logic is quite confusing (with the double negatives and coercion of booleans to integers). it might be nicer as
```
CASE WHEN
registered_status IS DISTINCT FROM 0 AND (agg_migration.is_migrated IS DISTINCT FROM 1 OR agg_migration.migration_date < %(start_date)s) THEN 1 ELSE 0 END
```
you can double check the logic but I think it is much clearer and more direct that way.

Reply:
this actually changed the logic,
Initially the logic is `migration_status IS DISTINCT FROM 1` but since migration_status is more of a calculated value now which is equal to `((agg_migration.is_migrated=1 AND
              agg_migration.migration_date < %(start_date)s)`

Reply:
Like I said, I am not positive of the logic, I think writing it without double negatives and integer coercion will be clearer. If it is that instead of mine that seems is fine and still clearer. I see where mine is wrong though. Perhaps 
```
CASE WHEN
registered_status IS DISTINCT FROM 0 AND (agg_migration.is_migrated IS DISTINCT FROM 1 OR agg_migration.migration_date >= %(start_date)s) THEN 1 ELSE 0 END
```
is the correct one. The overarching idea is that you are valid if you are registered and (not_migrated OR migration_date is after the current month)

Reply:
@calellowitz Got it, negation of A and B is not A OR not B, just confused by this `agg_migration.migration_date < %(start_date)s`

Reply:
Done

Reply:
You appear to still be coercing the boolean to an integer. Is there a reason for that?